### ISSUE TITLE
Make x264 an optional dependency?

### ISSUE THREAD

**Issue by MichaelAquilina** [ammen99/wf-recorder#28]
I am trying to add `wf-recorder` to openSUSEs build service so that is available from the package repository - but I am blocked by the fact that x264 is not available in the standard repositories.

In order to build it locally I need to add the external [Packman](https://ftp.gwdg.de/pub/linux/misc/packman/suse/openSUSE_Tumbleweed/Essentials/) repo.

Is it possible to make x264 an optional dependency an rely on other codecs to encode the videos recorded?

**Comment by ammen99** [comment]
The libx264 codec is the default one. We'd have this problem with any codec. But actually I'm not very sure that libx264 is actually needed for compilation. Could you test if just removing it from `meson.build` and using different codecs from the command line helps? If yes, I can just add an option to change the default codec.

**Comment by MichaelAquilina** [comment]
Seems like there are a bunch of `#include <x264.h>` statements in the code - so this seems like it is currently an actual dependency.

I believe the reason why x264 is not included in the default openSUSE repos is because its a non-open codec. If that's the case, other distros will probably have the same issue.

**Comment by ammen99** [comment]
@MichaelAquilina I think @soreau tested and could remove the includes. Also it doesn't look like libx264 is a direct dependency of Wayfire, I don't remember writing code that uses the library directly.

**Comment by MichaelAquilina** [comment]
@ammen99 you are right - removing x264.h compiles successfully:

```
diff --git a/meson.build b/meson.build
index 19a84d5..8e01653 100644
--- a/meson.build
+++ b/meson.build
@@ -23,11 +23,10 @@ libavcodec = dependency('libavcodec')
 libavformat = dependency('libavformat')
 sws = dependency('libswscale')
 swr = dependency('libswresample')
-x264 = dependency('x264')
 threads = dependency('threads')
 pulse = dependency('libpulse-simple')
 
 subdir('proto')
 executable('wf-recorder', ['src/frame-writer.cpp', 'src/main.cpp', 'src/pulse.cpp', 'src/averr.c'],
-        dependencies: [wayland_client, wayland_protos, libavutil, libavcodec, libavformat, wf_protos, x264, sws, threads, pulse, swr],
+        dependencies: [wayland_client, wayland_protos, libavutil, libavcodec, libavformat, wf_protos, sws, threads, pulse, swr],
         install: true)
diff --git a/src/frame-writer.hpp b/src/frame-writer.hpp
index 95c2e6e..f717145 100644
--- a/src/frame-writer.hpp
+++ b/src/frame-writer.hpp
@@ -13,7 +13,6 @@
 
 extern "C"
 {
-    #include <x264.h>
     #include <libswscale/swscale.h>
     #include <libswresample/swresample.h>
     #include <libavcodec/avcodec.h>
```

The binary seems to work without any issues too

**Comment by soreau** [comment]
@MichaelAquilina Can you test PR #29? It removes the x264 dependency and lets you choose a default codec by passing meson `-Ddefault_codec="libx264"` for example.

**Comment by ammen99** [comment]
Fixed in #29
